source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+
+### Wait on `GET` request matching `url`
+
+```javascript
+cy.route2('**/users').as('getUsers')
+cy.visit('/users')
+cy.wait('@getUsers')
+```
+
+### Wait on matching `method` and `url`
+
+```javascript
+cy.route2('POST', '**/users').as('postUser')
+cy.visit('/users')
+cy.get('#first-name').type('Julius{enter}')
+cy.wait('@postUser')
+```
+
+### Wait on `url` matching glob
+
+Under the hood Cypress uses {% url 'minimatch' https://github.com/isaacs/minimatch %} to match glob patterns of `url`.
+
+This means you can take advantage of `*` and `**` glob support. This makes it *much* easier to route against dynamic segments without having to build up a complex `RegExp`.
+
+We expose {% url `Cypress.minimatch` minimatch %} as a function that you can use in your console to test routes.
+
+### Match route against any UserId
+
+```javascript
+cy.route2('**/users/*/comments')
+
+// https://localhost:7777/users/123/comments     <-- matches
+// https://localhost:7777/users/123/comments/465 <-- does not match
+```
+
+### Use glob to match all segments
+
+```javascript
+cy.route2('**/posts/**')
+
+// https://localhost:7777/posts/1            <-- matches
+// https://localhost:7777/posts/foo/bar/baz  <-- matches
+// https://localhost:7777/posts/quuz?a=b&1=2 <-- matches
+```
+
+### Modify request
+
+Before a request gets passed through, we can modify the request before it is sent out through a function. The function exposes a `req` object which contains the request.
+
+You can also (optionally) add a simulated {% urlHash 'delay' Simulate-delay %} or {% urlHash 'throttle' Simulate-throttle %} on the request to be sent out.
+
+```javascript
+cy.route2('/users', (req) => {
+  // modify the headers or body
+  req.headers = {...req.headers, accept: 'application/json'}
+  req.body = {...req.body, foo: 'bar'}
+})
+```
+
+### Modify response
+
+After a request that gets passed through, we can modify the response from the server before it is returned through a function. The function exposes a `req` object which contains the request. The `req.send()` function exposes a `res` object which contains the response.
+
+```javascript
+cy.route2('/users', (req) => {
+  // passing a function to req.send causes the request to pass through
+  // and allows the response from the origin server to be modified
+  req.send((res) => {
+    res.status = 200
+    res.headers['x-new-headers'] = 'from-server'
+
+    // dynamically alter body
+    res.body = res.body.replace('window.top', 'window.self')
+  })
+})
+```
+
+## With Stubbing
+
+If you pass a `response` to `cy.route()`, Cypress will stub the response in the request.
+
+### `url` as a string
+
+When passing a `string` as the `url`, the URL must match *exactly* what you've written. You'll want to use the decoded string and not include any hash encoding (ie. use `@` instead of `%40`).
```

Comment:
not true, it is matched via `minimatch`. also check with tests that this is still true for cy.route2, i believe this was copied from cy.route